src/index.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { useEffect, useState, useRef } from 'react';
+
+type IWindowProps = {
```

Comment:
[nitpick] Why not use `interface` for this instead? I guess the I in the name is for Interface anyway :)